app/services/work_packages/shared/working_days.rb

```diff
@@ -54,11 +54,20 @@ def lag(predecessor_date, successor_date)
       def add_lag(date, lag)
         return nil unless date
 
-        date_after_lag = date + 1.day
         lag ||= 0
-        while lag > 0
-          lag -= 1 if working?(date_after_lag)
-          date_after_lag += 1
+        negative_lag = lag < 0
+
+        date_after_lag = date + 1.day
```

Comment:
I personally find the following reads a bit better:

```suggestion
        date_after_lag = date.next_day
```

or

```suggestion
        date_after_lag = date.succ
```

Reply:
`next_day` will be perfect.